client_test/rpc_test.go

```diff
@@ -44,143 +46,244 @@ func createLogger(logDestination *os.File, clientName, rpcRole string) zerolog.L
-		expectedLedgerInfo(bobResponse.ChannelId, simpleOutcome(ta.Bob.Address(), ta.Irene.Address(), 101, 99), query.Complete),
+
+	aliceClient.Pay(vabCreateResponse.ChannelId, 1)
+
+	vabClosure := aliceClient.CloseVirtual(vabCreateResponse.ChannelId)
+	for _, client := range clients {
+		<-client.ObjectiveCompleteChan(vabClosure)
 	}
-	checkNotifications(t, expectedBobLedgerNotifs, []query.LedgerChannelInfo{}, bobLedgerNotifs, defaultTimeout)
 
-	requiredVirtualNotifs := []query.PaymentChannelInfo{
-		expectedPaymentInfo(vRes.ChannelId, simpleOutcome(ta.Alice.Address(), ta.Bob.Address(), 100, 0), query.Proposed),
-		expectedPaymentInfo(vRes.ChannelId, simpleOutcome(ta.Alice.Address(), ta.Bob.Address(), 100, 0), query.Open),
-		expectedPaymentInfo(vRes.ChannelId, simpleOutcome(ta.Alice.Address(), ta.Bob.Address(), 99, 1), query.Open),
-		expectedPaymentInfo(vRes.ChannelId, simpleOutcome(ta.Alice.Address(), ta.Bob.Address(), 99, 1), query.Complete),
+	laiClosure := aliceClient.CloseLedger(aliceLedger.ChannelId)
+	<-aliceClient.ObjectiveCompleteChan(laiClosure)
+	// bobClient.CloseLedger(bobLedger.ChannelId)
+
+	if n != 2 { // for n=2, alice and bob share a ledger, which should only be closed once.
+		libClosure := bobClient.CloseLedger(bobLedger.ChannelId)
+		<-bobClient.ObjectiveCompleteChan(libClosure)
 	}
-	optionalVirtualNotifs := []query.PaymentChannelInfo{
-		expectedPaymentInfo(vRes.ChannelId, simpleOutcome(ta.Alice.Address(), ta.Bob.Address(), 99, 1), query.Closing),
-		// TODO: Sometimes we see a closing notification with the original balance.
-		// See https://github.com/statechannels/go-nitro/issues/1306
-		expectedPaymentInfo(vRes.ChannelId, simpleOutcome(ta.Alice.Address(), ta.Bob.Address(), 100, 0), query.Closing),
+
+	//////////////////////////
+	// perform wrap-up checks
+	//////////////////////////
+
+	for i, client := range clients {
+		if i != 0 {
+			leftLC := ledgerChannels[i-1]
+			vcCount := len(client.GetPaymentChannelsByLedger(leftLC.ChannelId))
+			if vcCount != 0 {
+				t.Errorf("expected no virtual channels in ledger channel %s, got %d", leftLC.ChannelId, vcCount)
+			}
+		}
+		if i != n-1 {
+			rightLC := ledgerChannels[i]
+			vcCount := len(client.GetPaymentChannelsByLedger(rightLC.ChannelId))
+			if vcCount != 0 {
+				t.Errorf("expected no virtual channels in ledger channel %s, got %d", rightLC.ChannelId, vcCount)
+			}
+		}
 	}
-	checkNotifications(t, requiredVirtualNotifs, optionalVirtualNotifs, aliceVirtualNotifs, defaultTimeout)
 
-	checkNotifications(t, requiredVirtualNotifs, optionalVirtualNotifs, bobVirtualNotifs, defaultTimeout)
+	aliceLedgerNotifs := aliceClient.LedgerChannelUpdatesChan(ledgerChannels[0].ChannelId)
+	expectedAliceLedgerNotifs := createLedgerStory(
+		aliceLedger.ChannelId, alice.Address(), actors[1].Address(), // actor[1] is the first intermediary - can be Bob if n=2 (0-hop)
+		[]channelStatusShorthand{
+			{100, 100, query.Proposed},
+			{100, 100, query.Open},
+			{0, 100, query.Open},
```

Comment:
I don't think it's the case, but is this the order we expect? I don't think we check the order of notification in `checkNotifications`, but this is still a bit confusing.

Reply:
This is the correct order - the shorthand is not a complete description, so it's not as obvious as it could be.

- A:100, I:100
- A:0, I:100, G(vAB): 100  (Alice forwards 100 to the guarantee)
- A: 99, I: 101 (After vAB closes, A and I's balances adjust)

Reply:
... But in any case the test itself is order insensitive for this data.

Reply:
fc952786